File: 2024/56. Merge Intervals/Solution2.java
class Solution {
    public int[][] merge(int[][] intervals) {
        if (intervals == null || intervals.length < 2)
            return intervals;
        List<int[]> list = new ArrayList<>();
        Arrays.sort(intervals, (int[] x, int[] y) -> (x[0] - y[0]));
        int i = 0, j = 1;
        while (j <= intervals.length) {
            if (j == intervals.length || intervals[j][0] > intervals[i][1]) {
                list.add(new int[]{intervals[i][0], intervals[i][1]});
                i = j;
            } else {
                intervals[i][1] = Math.max(intervals[j][1], intervals[i][1]);
            }
            j++;
        }
        return list.toArray(new int[list.size()][2]);
    }
}

File: 2025/123. Best Time to Buy and Sell Stock III/solution2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        dp = [[[0] * 2 for _ in range(3)] for _ in range(len(prices) + 1)]
        for i in range(len(prices) - 1, -1, -1):
            for r in range(1, 3):
                dp[i][r][0] = max(-prices[i] + dp[i + 1][r][1], dp[i + 1][r][0])
                dp[i][r][1] = max(prices[i] + dp[i + 1][r - 1][0], dp[i + 1][r][1])
        return dp[0][2][0]